### ISSUE TITLE
Allow a non-logged-in viewer of a share-by-identity link to click on an email token card to log in

### ISSUE THREAD

**Issue by paulproteus** [sandstorm-io/sandstorm#1568]
The problem:

- If you're a logged-out user, and
- Someone share-by-identity-s with you, and
- That identity is an email token identity, then
- You receive an email that takes you to a page with an identity card, and
- That identity card can't be clicked.

Proposal:

- The identity card can be clicked, and
- When you click it, it sends a login email to the email address, and
- It triggers the "Sign in" drop-down, which is in the state of "Enter the token",
- And for clarity, the email icon in the sign-in drop down is dark (`:hover` state even if the sign in drop-down isn't being `:hover`'d)

**Comment by paulproteus** [comment]
@dwrensha seemed OK with this in an out-of-band conversation. @neynah also approves it.

**Comment by zarvox** [comment]
I agree this would be an improvement over the current state of things, and I am in favor of the proposed change.

But.

It feels kinda silly that you receive an email with the link, follow the link, and then you need to prove ownership of the email that you just got the link from, which requires that you go back to the inbox that you literally just came from, which is the primary source from which you could have obtained that link to begin with.  It is rather unfortunate that opening a shared link like this involves so many back-and-forths - this piece of friction will be confusing and surprising to users.

I really wish we could come up with a solution for this that avoids that second trip to the email inbox.  I understand the security arguments against the sharing link carrying login power (users might forward it to others, we don't really want to support long-lived tokens), but I hope there's some middle ground we can find that can at least improve the likely-common case of "I just got this sharing link in an email and want to use it, now I need a second email to do anything with the first".

**Comment by kentonv** [comment]
Note that this issue isn't specific to email login. Clicking on a Google or Github identity should initiate the corresponding login flow. Note that Google even has a way to specify which Google account to authenticate, which we could use here (Github does not AFAIK).
